controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +134,335 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+			var storageClassClaimOwnerReference metav1.OwnerReference
+			for _, resource := range externalResources {
+
+				// storageClassClaimOwnerReference points to the StorageClassClaim object as the owner object of the created resource(s).
+				storageClassClaimOwnerReference = metav1.OwnerReference{
+					APIVersion: r.storageClassClaim.APIVersion,
+					Kind:       r.storageClassClaim.Kind,
+					Name:       r.storageClassClaim.Name,
+					UID:        r.storageClassClaim.UID,
+				}
+				switch resource.Kind {
+				case "Secret":
+					objectMeta := metav1.ObjectMeta{
+						Name:      resource.Name,
+						Namespace: r.storageClassClaim.Namespace,
+						OwnerReferences: []metav1.OwnerReference{
+							storageClassClaimOwnerReference,
+						},
+					}
+					secret := &corev1.Secret{
+						ObjectMeta: objectMeta,
+						Data:       make(map[string][]byte),
+					}
+					for key, value := range resource.Data {
+						secret.Data[key] = []byte(value)
+					}
+					secretTemplate := &corev1.Secret{ObjectMeta: objectMeta}
+					err := r.get(secretTemplate)
+					if err != nil {
+						if errors.IsNotFound(err) {
+							r.log.Info("Creating a new Secret", "Secret", secret.Name)
+							if err := r.Client.Create(r.ctx, secret); err != nil {
+								return reconcile.Result{}, fmt.Errorf("failed to create Secret: %v", err)
+							}
+						} else {
+							return reconcile.Result{}, fmt.Errorf("failed to get Secret: %v", err)
+						}
+					} else if !equality.Semantic.DeepEqual(secret.Data, secretTemplate.Data) {
+						r.log.Info("Updating Secret", "Secret", secretTemplate.Name)
+						secretTemplate.Data = secret.Data
+						if err := r.Client.Update(r.ctx, secretTemplate); err != nil {
+							return reconcile.Result{}, fmt.Errorf("failed to update Secret: %v", err)
+						}
+					}
+				case "StorageClass":
+					var storageClassConfiguration StorageClassConfiguration
+					name := resource.Name
+					if name == "cephfs" {
+						storageClassConfiguration = r.newCephFilesystemStorageClassConfiguration(storageClassConfiguration)
+						enableRookCSICephFS = true
+					} else if name == "ceph-rbd" {
+						storageClassConfiguration = r.newCephBlockPoolStorageClassConfiguration(storageClassConfiguration)
+					}
+					for key, value := range resource.Data {
+						storageClassConfiguration.storageClass.Parameters[key] = value
+					}
+					availableStorageClassConfigurations = append(availableStorageClassConfigurations, storageClassConfiguration)
+				}
+			}
+			err = r.createStorageClasses(availableStorageClassConfigurations, storageClassClaimOwnerReference)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to create StorageClass: %v", err)
+			}
+			err = r.setRookCSICephFS(enableRookCSICephFS)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to set RookEnableCephFSCSIKey to EnableRookCSICephFS [%v]: %s", enableRookCSICephFS, err)
+			}
```

Comment:
Why wait with this call till the end of the func? 
The natural place to do this is in the StorageClass case itself when handling the cephfs subcase.
If you do that then there will be no need to "remember" that you have a cephfs case in the first place which will remove then need for the `enableRookCSICephFS` var